packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -361,11 +354,15 @@ function useRestoreFocus(scopeRef: RefObject<HTMLElement[]>, restoreFocus: boole
         e.preventDefault();
         e.stopPropagation();
         if (nextElement) {
-          nextElement.focus();
+          focusElement(nextElement, true);
         } else {
           // If there is no next element, blur the focused element to move focus to the body.
           focusedElement.blur();
         }
+      } else if (nextElement && !isElementInScope(nextElement, scope)) {
```

Comment:
It would, but we I'm not sure we can unit test that behavior. I think this was added to fix a unit test failure.

Reply:
See test at https://github.com/adobe/react-spectrum/pull/1493/files/b9e06919bd772a844edb55d6342696b5940448e5#diff-4ede2be95e8bcfd65b5421b9ae5394aa17e3fc11f77217b9088166f655690007R832-R858

Reply:
Ah ok. You might be able to try `userEvent.tab()` instead of firing the keyboard event, which should simulate the browser's default behavior more fully.